### PR TITLE
Remove some builds from Travis-CI matrix

### PR DESCRIPTION
**Pull request by pitrou**
Since Travis-CI is a limited public infrastructure, strive not to test too many combinations.

### CODE REVIEW

**Comment by pitrou** [comment]
@mrocklin

**Comment by mrocklin** [comment]
CC @jakirkham who I think added the newer numpy builds.  Any objection to this change?

**Comment by jakirkham** [comment]
Completely agree we should trim the matrix. Personally I would push for dropping NumPy 1.10 as opposed to 1.11. Though maybe I'm not understanding appeal for keeping 1.10 in so many places.

**Comment by pitrou** [comment]
I suppose some people are still using Numpy 1.10? @mrocklin, what do you think?

**Comment by jcrist** [comment]
A few data points:

- Only version checks we have in `dask.array` are for `numpy < 1.10`, so there's no known breakage at least
- Scikit-image (a dependent package) requires `numpy >= 1.11`
- Xarray (also a dependent) requires `numpy >= 1.7`

**Comment by mrocklin** [comment]
I think that there is some value to testing against a wide range of numpy
versions.

For a library, if we support version 1 and version 10 then is it somewhat
likely that we support version 5.

**Comment by jakirkham** [comment]
To be clear, it seems fine that NumPy 1.10 should be tested in the CI. Would just expect more testing effort be applied on more recent versions of NumPy and less effort on old versions of NumPy. In fact, I would think this would mirror usage as well (perhaps a justification of this shift in effort).

File: .travis.yml [hunk 1]
```diff
@@ -5,11 +5,9 @@ os: linux
 env:
   matrix:
     - PYTHON=2.7 NUMPY=1.10.4 PANDAS=0.19.0 COVERAGE='true' PARALLEL='false' TEST_IMPORTS='false' XTRATESTARGS=
-    - PYTHON=2.7 NUMPY=1.11.0 PANDAS=0.19.2 COVERAGE='false' PARALLEL='true' TEST_IMPORTS='false' PYTHONOPTIMIZE=2 XTRATESTARGS=--ignore=dask/diagnostics
```
Comment (jakirkham) [inline .travis.yml]: To clarify what I'm thinking, if it were up to me, I'd keep this line, but drop 1.10.4 above it.

File: .travis.yml [hunk 2]
```diff
@@ -5,11 +5,9 @@ os: linux
...
     - PYTHON=2.7 NUMPY=1.12.1 PANDAS=0.19.2 COVERAGE='false' PARALLEL='true' TEST_IMPORTS='false' PYTHONOPTIMIZE=2 XTRATESTARGS=--ignore=dask/diagnostics
     - PYTHON=3.4 NUMPY=1.10.4 PANDAS=0.19.1 COVERAGE='false' PARALLEL='true' TEST_IMPORTS='false' PYTHONOPTIMIZE=2 XTRATESTARGS=--ignore=dask/diagnostics
     - PYTHON=3.5 NUMPY=1.11.0 PANDAS=0.19.2 COVERAGE='false' PARALLEL='true' TEST_IMPORTS='false' XTRATESTARGS=
-    - PYTHON=3.6 NUMPY=1.11.2 PANDAS=0.19.2 COVERAGE='false' PARALLEL='true' TEST_IMPORTS='true' XTRATESTARGS=
```
Comment (jakirkham) [inline .travis.yml]: Could live with this line being dropped, but maybe it is time to bump the patch version of NumPy for the Python 3.5 build. Unless there is a reason not to.

**Comment by mrocklin** [comment]
In general I'm in support of this PR as written.  I don't think we need to worry too much about numpy version coverage.  We use NumPy in a fairly mundane way.  I'm more worried about making sure that we support older versions where fewer features are supported than in covering the more recent (but not most recent) versions.

**Pull request merged**